### ISSUE TITLE
Unable to generate manifest for ESM based CLI Tool

### ISSUE THREAD

**Issue by tonyt-adept** [oclif/oclif#1117]
Do you want to request a *feature* or report a *bug*?
-----------------------------------------------------
Bug/Unexpected Behavior 
This could very well be a niche issue caused by a hybrid ESM/CJS Typescript project

What is the current behavior?
-----------------------------
Running `oclif manifest` results in:
```
ModuleLoadError Plugin: <my-cli-name>: [MODULE_NOT_FOUND] import() failed to load <transpiled-command>.js: Cannot find module '<local module import>' imported from
 ›   <transpiled-command>.js
```

I would expect `oclif` to properly update my manifest file.

In order to get my CLI tool running as `type: module` - I followed the steps laid out here: https://github.com/oclif/core/issues/421

My Typescript CLI project declares: `"type": "module"` in `package.json`, and the relevant `tsconfig.json` values are as follows:
```{
  "compilerOptions": {
    "esModuleInterop": true,
    "target": "ES2020",
    "module": "ES2020",
    "lib": ["ES2020"],
    "moduleResolution": "node"
  },
  "ts-node": {
    "esm": true,
    "experimentalSpecifierResolution": "node"  
    },
  "include": ["..."]
}
```
Following these steps allows my CLI tool to run as expected (invoking `/bin/run.js`) - however I'd like to ship an updated manifest file - which I previously did via `oclif manifest` but now errors (after moving to ESM).

My best sleuthing leads me to believe that the Dynamic Import causing the `MODULE_NOT_FOUND` error above [thrown here?](https://github.com/oclif/core/blob/main/src/module-loader.ts#L48) is not using the same Module Resolution strategies that were added to my CLI's `/bin/run.js` via the above link.

I'm not blaming `oclif manifest` or saying I'd expect this to work - I just wanted to open this Issue and provide my workaround which was using `ts-node oclif manifest` to successfully generate the Manifest.

Hope this helps anyone else bashing their head against the wondrous landscape of mixing CJS & ESM.

Also shoutout to `oclif`, I love the tool!

------------------------------
Please mention your oclif and OS version.
I recently migrated my CLI from `@oclif/core ~1.3.0` and `oclif ~2.4.0` to the lastest (oclif/corev2 & oclif 3.7.0)

**Comment by its-miller-time** [comment]
I am experiencing the same issue

**Comment by gotjoshua** [comment]
> my workaround which was using `ts-node oclif manifest` to successfully generate the Manifest.

i tried this, but got
```
$ ts-node oclif manifest
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module './oclif'
```

any help?

**Comment by gotjoshua** [comment]
i managed to get the manifest building by changing all directory imports into file.js imports...

so
```ts
import { getName } from '../../utils/file-utils.js' // instead of '../../utils/file-utils'
import Prep from '../prep/index.js' // instead of '../prep'
```
maybe it helps someone until ESM support is more solid

**Comment by mdonnalley** [comment]
Fix for this is now in `3.13.1-beta.0`
